tests/end-to-end/compass_e2e_test.go

```diff
@@ -1,14 +1,993 @@
+	assert.Equal(t, *in.Apis[1].Spec.FetchRequest.Mode, actReviewsApi.Spec.FetchRequest.Mode)
+	assert.Equal(t, in.Apis[1].Spec.FetchRequest.Filter, actReviewsApi.Spec.FetchRequest.Filter)
+	assert.NotNil(t, actReviewsApi.Spec.FetchRequest.Auth)
+
+}
+
+func TestCreateApplicationWithEventAPIs(t *testing.T) {
+	t.SkipNow()
+	// GIVEN
+	ctx := context.Background()
+	in := graphql.ApplicationInput{
+		Name: "wordpress",
+		EventAPIs: []*graphql.EventAPIDefinitionInput{
+			{
+				Name:        "comments/v1",
+				Description: ptrString("comments events"),
+				Version:     fixDepracatedVersion1(),
+				Group:       ptrString("comments"),
+				Spec: &graphql.EventAPISpecInput{
+					EventSpecType: graphql.EventAPISpecTypeAsyncAPI,
+					Data:          ptrCLOB(graphql.CLOB([]byte(`asyncapi: \"1.0.0\"`))),
+				},
+			},
+			{
+				Name:        "reviews/v1",
+				Description: ptrString("review events"),
+				Spec: &graphql.EventAPISpecInput{
+					EventSpecType: graphql.EventAPISpecTypeAsyncAPI,
+					FetchRequest: &graphql.FetchRequestInput{
+						URL:    "http://mywordpress.com/events",
+						Mode:   ptrFetchMode(graphql.FetchModePackage),
+						Filter: ptrString("async.json"),
+						Auth:   fixOauthAuth(),
+					},
+				},
+			},
+		},
+	}
+
+	appInputGQL, err := tc.graphqlizer.ApplicationInputToGQL(in)
+	require.NoError(t, err)
+
+	actualApp := ApplicationExt{}
+	createResp := resultMapperFor(&actualApp)
+	// WHEN
+	request := gcli.NewRequest(
+		fmt.Sprintf(
+			`mutation {
+  		result: createApplication(in: %s) { %s }}`,
+			appInputGQL,
+			tc.gqlFieldsProvider.ForApplication(),
+		))
+	saveQueryInExamples(t, request.Query(), "create application with event APIs")
+	err = tc.cli.Run(ctx, request, &createResp)
+	// THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, actualApp.ID)
+	defer deleteApplication(t, actualApp.ID)
+
+	assert.Len(t, actualApp.EventAPIs.Data, 2)
+	actCommentsApi := actualApp.EventAPIs.Data[0]
+	assert.NotNil(t, actCommentsApi.ID)
+	assert.Equal(t, in.EventAPIs[0].Name, actCommentsApi.Name)
+	assert.Equal(t, in.EventAPIs[0].Description, actCommentsApi.Description)
+	assert.Equal(t, in.EventAPIs[0].Group, actCommentsApi.Group)
+	assert.NotNil(t, actCommentsApi.Version)
+	assert.NotNil(t, actCommentsApi.Spec)
+
+	assert.Equal(t, in.EventAPIs[0].Spec.EventSpecType, actCommentsApi.Spec.Type)
+	assert.Equal(t, in.EventAPIs[0].Spec.Data, actCommentsApi.Spec.Data)
+
+	actReviewsApi := actualApp.EventAPIs.Data[1]
+	require.NotNil(t, actReviewsApi.Spec.FetchRequest)
+	assert.Equal(t, in.EventAPIs[1].Spec.FetchRequest.URL, actReviewsApi.Spec.FetchRequest.URL)
+	assert.Equal(t, in.EventAPIs[1].Spec.FetchRequest.Mode, actReviewsApi.Spec.FetchRequest.Mode)
+	assert.Equal(t, in.EventAPIs[1].Spec.FetchRequest.Filter, actReviewsApi.Spec.FetchRequest.Filter)
+	assert.NotNil(t, actReviewsApi.Spec.FetchRequest.Auth)
+
+}
+
+func TestCreateApplicationWithDocuments(t *testing.T) {
```

Comment:
I think there is too much detail in this test function. I would recommend moving some code parts to helper functions, like doing query ('when') and comparing results ('then'). It will make the whole test a lot easier to read.

Reply:
be specific, how you want to simplify this test.
- applicationInput is specific for this test, so IMO it does not make sense to create helper method for it, because it will be used only once. 
- creation of request is also specific only for this test
- the same with assertions

Reply:
assertions moved to the separate file